### PR TITLE
[SPARK-6909][SQL] Remove Hive Shim code

### PR DESCRIPTION
**Pull request by piaozhexiu**
This is a follow-up on #6393. I am removing the following files in this PR.

```
./sql/hive/v0.13.1/src/main/scala/org/apache/spark/sql/hive/Shim13.scala
./sql/hive-thriftserver/v0.13.1/src/main/scala/org/apache/spark/sql/hive/thriftserver/Shim13.scala
```

Basically, I re-factored the shim code as follows-
- Rewrote code directly with Hive 0.13 methods, or 
- Converted code into private methods, or
- Extracted code into separate classes

But for leftover code that didn't fit in any of these cases, I created a HiveShim object. For eg, helper functions which wrap Hive 0.13 methods to work around Hive bugs are placed here.

### CODE REVIEW

**Comment by marmbrus** [comment]
ok to test

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLSessionManager.scala [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    getAncestorField[Log](this, 3, "LOG").info(
+      s"HiveServer2: Async execution pool size $backgroundPoolSize")
+
+    setSuperField(this, "operationManager", sparkSqlOperationManager)
+    addService(sparkSqlOperationManager)
+
+    initCompositeService(hiveConf)
+  }
+
+  override def openSession(protocol: TProtocolVersion,
```
Comment (rxin) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLSessionManager.scala]: note for the future: we should indent this 4 space. i will fix it during merge.

File: sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala [hunk 2]
```diff
@@ -29,12 +26,15 @@ import org.apache.hive.service.server.{HiveServer2, ServerOptionsProcessor}
...
 import org.apache.spark.scheduler.{SparkListener, SparkListenerApplicationEnd, SparkListenerJobStart}
 import org.apache.spark.sql.SQLConf
+import org.apache.spark.sql.hive.HiveContext
 import org.apache.spark.sql.hive.thriftserver.ReflectionUtils._
 import org.apache.spark.sql.hive.thriftserver.ui.ThriftServerTab
-import org.apache.spark.sql.hive.{HiveContext, HiveShim}
 import org.apache.spark.util.Utils
 import org.apache.spark.{Logging, SparkContext}
 
+import scala.collection.mutable
```
Comment (rxin) [inline sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/HiveThriftServer2.scala]: note: import out of order. i will fix it in this case.

**Comment by rxin** [comment]
Thanks. I've merged this and will fix the minor issues.

**Comment by piaozhexiu** [comment]
@rxin Thank you! I'll keep in mind style nits next time.

**Pull request closed without merging**